packages/flutter/lib/src/services/asset_manifest.dart

```diff
@@ -2,25 +2,37 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:convert';
+
 import 'package:flutter/foundation.dart';
 
 import 'asset_bundle.dart';
 import 'message_codecs.dart';
 
 // We use .bin as the extension since it is well-known to represent
-// data in some arbitrary binary format. Using a well-known extension here
-// is important for web, because some web servers will not serve files with
-// unrecognized file extensions by default.
-// See https://github.com/flutter/flutter/issues/128456.
+// data in some arbitrary binary format.
 const String _kAssetManifestFilename = 'AssetManifest.bin';
 
+// We use the same bin file for the web, but re-encoded as JSON(base64(bytes))
+// so it can be downloaded by even the dumbest of browsers.
+// See https://github.com/flutter/flutter/issues/128456
+const String _kAssetManifestWebFilename = 'AssetManifest.bin.json';
+
 /// Contains details about available assets and their variants.
 /// See [Resolution-aware image assets](https://docs.flutter.dev/ui/assets-and-images#resolution-aware)
 /// to learn about asset variants and how to declare them.
 abstract class AssetManifest {
   /// Loads asset manifest data from an [AssetBundle] object and creates an
   /// [AssetManifest] object from that data.
   static Future<AssetManifest> loadFromAssetBundle(AssetBundle bundle) {
+    if (kIsWeb) {
```

Comment:
nit: a couple of lines (or links) above the `if` explaining how/why things are different on the web would be useful to future contributors.

Reply:
(Done)